active/CornerRadius/CornerRadiusToThickness2.md

```diff
@@ -0,0 +1,108 @@
+> See comments in Markdown for how to use this spec template
+
+
+# Background
+
+_This spec updates the existing [CornerRadiusToThicknessConverter](https://docs.microsoft.com/uwp/api/Microsoft.UI.Xaml.Controls.Primitives.CornerRadiusToThicknessConverter) class with new filters._
+
+WinUI has a [CornerRadius](https://docs.microsoft.com/uwp/api/Windows.UI.Xaml.CornerRadius) struct that's used to define the rounding the 4 corners of a box, and the [Thickness](https://docs.microsoft.com/uwp/api/Windows.UI.Xaml.Thickness) struct that's used for defining the padding or margin around the 4 sides of a box.
+
+CornerRadius defines the radius length of the circle segment that's used to draw a rounded corner. For example the following uses 4 different radii for the 4 corners (the corners are listed in clockwise order starting from the top left corner):
+
+```xml
+<Border CornerRadius="0,20,40,60" BorderBrush="Red" BorderThickness="2" Height="100" Width="100" />
+```
+
+![Corner radius example](./CornerRadius.png)
+
+Thickness defines 4 sides. For example the following uses 4 different values for the margins around a blue rectangle in a red box (the sides are listed in clockwise order starting from the left):
+
+```xml
+<Border BorderBrush="Red" BorderThickness="2" Height="100" Width="100" >
+    <Rectangle Fill="Blue" Margin="0,20,40,60" />
+</Border>
+```
+
+![Thickness example](./Thickness.png)
+
+For consistency it's useful to be able to build a Thickness out of components of a CornerRadius. For this reason the [CornerRadiusFilterConverter](https://docs.microsoft.com/uwp/api/Microsoft.UI.Xaml.Controls.Primitives.CornerRadiusFilterConverter) exists, which can be used in Xaml markup. That converter has a [CornerRadiusFilterKind](https://docs.microsoft.com/uwp/api/Microsoft.UI.Xaml.Controls.Primitives.CornerRadiusFilterKind) that lets you define which parts to pull out, what's being added in this spec is update to that enum to allow more parts to be pulled out.
+
+**Existing values:**
+* FilterTopAndBottomFromLeft,
+* FilterTopAndBottomFromRight,
+* FilterLeftAndRightFromTop,
+* FilterLeftAndRightFromBottom
+
+**New values:**
+* FilterTopFromTopLeft,
+* FilterTopFromTopRight,
+* FilterRightFromTopRight,
+* FilterRightFromBottomRight,
+* FilterBottomFromBottomRight,
+* FilterBottomFromBottomLeft,
+* FilterLeftFromBottomLeft,
+* FilterLeftFromTopLeft,
+
+The CornerRadiusFilterConverter is also updated with a new property -- Multiplier -- which is applied during the converstion. For example, setting this to -1 negates the value.

```

Comment:
1. `conversion` instead of `converstion`
2. The new multiplier property should also be present in the API details down
3. Maybe Multiplier could be of type thickness, allowing scaling different parts by different factors. Maybe we could also allow "FilterAllToAll" (semantically speaking) to make the converter even more powerful.

Reply:
What would FilterAllToAll do? Both structs have 4 fields, but really we're mapping 8 values onto 4.

Reply:
FilterAllToAll would do something like this:

![image](https://user-images.githubusercontent.com/16122379/86976416-afedf580-c17a-11ea-91b4-7979a02b667e.png)

We could also do the max/min/average of the corner radii on a side to get the thickness for that side, e.g. `Left = min(BottomLeft,TopLeft)`.

Reply:
Those work but I can't see the use case for them, for example when a UI would want to align the top margin of one element with the top/left corner radius of another element. I think we should keep what you have and add more when we need it.

Reply:
Fair point, it would definitely not be a common case. Let's not add it then and stick to what is in this proposal already.